ee/app/livechat-enterprise/server/business-hour/Multiple.ts

```diff
@@ -0,0 +1,221 @@
+import moment from 'moment';
+
+import {
+	AbstractBusinessHour,
+	IBusinessHour,
+} from '../../../../../app/livechat/server/business-hour/AbstractBusinessHour';
+import { ILivechatBusinessHour, LivechatBussinessHourTypes } from '../../../../../definition/ILivechatBusinessHour';
+import { LivechatDepartment, LivechatDepartmentAgents } from '../../../../../app/models/server/raw';
+import { LivechatDepartmentRaw } from '../../../../../app/models/server/raw/LivechatDepartment';
+import { LivechatDepartmentAgentsRaw } from '../../../../../app/models/server/raw/LivechatDepartmentAgents';
+
+interface IBusinessHoursExtraProperties extends ILivechatBusinessHour {
+	timezoneName: string;
+	departmentsToApplyBusinessHour: string;
+}
+
+export class MultipleBusinessHours extends AbstractBusinessHour implements IBusinessHour {
+	private DepartmentsRepository: LivechatDepartmentRaw = LivechatDepartment;
+
+	private DepartmentsAgentsRepository: LivechatDepartmentAgentsRaw = LivechatDepartmentAgents;
+
+	async closeBusinessHoursByDayAndHour(day: string, hour: string): Promise<void> {
+		const businessHours = await this.BusinessHourRepository.findActiveBusinessHoursToClose(day, hour, undefined, {
+			fields: {
+				_id: 1,
+				type: 1,
+			},
+		});
+		for (const businessHour of businessHours) {
+			this.closeBusinessHour(businessHour);
+		}
+	}
+
+	async getBusinessHour(id: string): Promise<ILivechatBusinessHour | undefined> {
+		if (!id) {
+			return;
+		}
+		const businessHour: ILivechatBusinessHour = await this.BusinessHourRepository.findOneById(id);
+		businessHour.departments = await this.DepartmentsRepository.findByBusinessHourId(businessHour._id, { fields: { name: 1 } }).toArray();
+		return businessHour;
+	}
+
+	async openBusinessHoursByDayHour(day: string, hour: string): Promise<void> {
+		const businessHours = await this.BusinessHourRepository.findActiveBusinessHoursToOpen(day, hour, undefined, {
+			fields: {
+				_id: 1,
+				type: 1,
+			},
+		});
+		for (const businessHour of businessHours) {
+			this.openBusinessHour(businessHour);
+		}
+	}
+
+	async saveBusinessHour(businessHourData: IBusinessHoursExtraProperties): Promise<void> {
+		businessHourData.timezone = {
+			name: businessHourData.timezoneName,
+			utc: this.getUTCFromTimezone(businessHourData.timezoneName),
+		};
+		if (businessHourData.timezone.name) {
```

Comment:
It's possible to improve this long block of code(if/else), reducing the complexity.